src/transformers/generation_tf_utils.py

```diff
@@ -2195,6 +2204,45 @@ def greedy_search(
         else:
             return input_ids
 
+    @tf.function(jit_compile=True)
+    def xla_greedy_generate(self, input_ids, max_length):
+        model_inputs = self.new_prepare_inputs_for_generation(input_ids, max_length=max_length)
+        past = model_inputs["past_key_values"]
+        generated = model_inputs["generated"]
+        next_tokens = model_inputs["next_tokens"]
+        transposed_attention_mask = model_inputs["transposed_attention_mask"]
+        batch_size, seq_length = input_ids.shape
+
+        ptr = tf.ones(shape=(1,), dtype=tf.int32)
+        max_tokens_to_generate = max_length - seq_length
+
+        def greedy_search_cond_fn(input_ids, transposed_attention_mask, next_tokens, past, ptr):
+            """state termination condition fn."""
+            return ptr < max_tokens_to_generate
```

Comment:
Can you check here in a hacky way whether there is an EOS in `input_ids` to stop?

Reply:
Renamed to `current_pos`!